inst/pages/Introductory-workflow/Introductory.qmd

```diff
@@ -0,0 +1,293 @@
+  plot_layout(guides = "collect")
+
+```
+
+It's very important to make all these comparisons in order to quantify diversity and compare samples in our data using different measures. - You can find other types of comparisons directly in the book @sec-community-diversity.
+
+<hr>
+
+## Community similarity
+
+Community similarity refers to the way microorganisms resemble each other in terms of their composition and abundance of different microbial taxa. This can help us understand to what degree different samples resemble each other and finding key information. In microbiome analysis however, it's more common to measure the dissimilarity/Beta diversity between two samples A and B using the bray-Curtis measure which is defined as follows:
+
+$$
+BC_{ij} = \frac{\sum_{k} |A_{k} - B_{k}|}{\sum_{k} (A_{k} + B_{k})}
+$$ Luckily for us, the [**mia package**](https://microbiome.github.io/mia/) provides an easy way to calculate the Bray-Curtis dissimilarity and add it to our TreeSE using the `transformAssay` method:
+
+```{r}
+tse <- transformAssay(tse,
+                      assay.type = "counts",
+                      method = "relabundance")
+```
+
+This will take the original counts assay and apply the Bray-Curtis formula to it. The result is a matrix with the sample identifiers as rows and the dissimilarity to other samples as columns. It can be accessed in the assays of the tse:
+
+```{r}
+assay(tse, "relabundance")[5:10,1:10]
+```
+
+In our case, the assay contains 151 rows and 27 columns. Having so many columns and thus dimensions can be troublesome for visualizing dissimilarity.
+
+In order to visualize the dissimilarity between the different samples, we can conduct a Principal Coordinate analysis on the newly created assay. This essentially projects the Bray-curtis dimensions onto a lower space whilst maintaining as much of the variation as possible, the projected values are called principal coordinates. You can read more [**here**](https://en.wikipedia.org/wiki/Multidimensional_scaling#Types).
+
+MIA doesn't provide a direct way to reduce dimensionality but we can use Bioconductor's scater package and the vegan package, the latter can be found [**here**](https://cran.r-project.org/web/packages/vegan/vegan.pdf) to turn the dissimilarity into actual distances that can be visualized:
+
+```{r}
+#| output: false
+# Load package to plot reducedDim
+library(scater)
+
+# Run PCoA on relabundance assay with Bray-Curtis distances
+tse <- runMDS(tse,
+              FUN = vegan::vegdist,
+              method = "bray",
+              assay.type = "relabundance",
+              name = "MDS_bray")
+```
+
+Now that we calculated the principal coordinates we can plot them onto a two dimensional space as such:
+
+```{r}
+#| output: false
+# Makes the plots interactive
+library(plotly)
+```
+
+```{r}
+# Create ggplot object
+p <- plotReducedDim(tse, "MDS_bray") +
+  geom_point(color = "#659b91")
+
+# Convert to an interactive plot with ggplotly
```

Comment:
You could show 3 dimensional plot since you are using plotly

Reply:
Also might be good example to who how color those points with variables from colData

Reply:
We discussed this with Leo and Noah has moved the 3d version of this plot to OMA so that we can keep an easier 2d version in the workflow.

Reply:
Should I still color the points ?

Reply:
You could colour the based on sample metadata.

Does ggplotly give any additional value?

Reply:
It adds interactivity to the plot so people can actually look at the individual values.